driver/utils.py

```diff
@@ -91,7 +99,24 @@ def fetch_job_logs(job_name, namespace, tests_local_run):
     subprocess.check_call(command)
 
 
-def delete_job(job_name, namespace, lightkube_client=None):
-    """Delete a Kubernetes Job."""
-    client = lightkube_client or Client(trust_env=False)
-    client.delete(Job, name=job_name, namespace=namespace)
+@tenacity.retry(
+    wait=tenacity.wait_exponential(multiplier=2, min=1, max=10),
+    stop=tenacity.stop_after_attempt(10),
+    reraise=True,
+)
+def assert_profile_deleted(client, profile_name, logger: logging.Logger):
+    """Assert that the Profile is deleted.
+
+    Retries multiple times to allow for the Profile to be deleted.
+    """
+    logger.info(f"Waiting for Profile {profile_name} to be deleted.")
+    deleted = False
+    try:
+        client.get(PROFILE_RESOURCE, profile_name)
+    except ApiError as error:
+        logger.info(f"Unable to get Profile {profile_name} (status: {error.status.code})")
```

Comment:
Should we have this under the `if` statement? If users see this log, it looks like something unexpected happen, while we 're waiting to assert that this has been deleted. This way, we will only see it if the error is not 404.